### PR TITLE
Allow for setting reviwers per author

### LINKED ISSUE #25: Allow for specifying reviewers for each member
The current implementation is best for the teams where all members review all PRs.
However, in some teams, the following case can happen, for instance:

* Head of Design reviews only Lead Designer's changes
* Lead Designer reviews Head of Design's, Member A's and B's changes
* Member A reviews Lead Designer's changes
* Member B doesn't review changes at all

In other words,

* Member A has Lead Designer review their code
* Member B has Lead Designer review their code
* Lead Designer has Member A and Head of Design review their code

| reviewer \ reviewee | Head of Design | Lead Designer | Member A | Member B |
| --- |  --- | --- | --- | --- |
|  Head of Design | - | Y | N | N |
| Lead Designer | Y | - | Y | Y |
| Member A | N | Y | - | N |
| Member B | N | N | N | - |

## Solution
Add a `per_author` key to `reviewers`. The settings file will look like:

```yaml
reviewers:
  defaults:
    - engineers

  groups:
    engineers:
      - engineer_a
      - engineer_b
      - engineer_c
    designers:
      - head_of_design
      - lead_designer
      - desinger_a
      - designer_b

  # Reviwers for each author
  per_author:
    engineers:
      - engineers
    head_of_design:
      - lead_designer
    lead_designer:
      - head_of_design
      - designer_a
    designer_a:
      - lead_designer
    designer_b:
      - lead_desinger
```

### PR DESCRIPTION
**Pull request by necojackarc**
Resolve #25

Add `per_author` setting to `reviewers` to allow for specifying reviewers for each author.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,8 @@ It varies depending on the team who should review which pull requests. In some t
 This GitHub Action best suits any of the following needs:
 
 - You'd like to request review based on files changed
-- You'd like to request review to all of the other team members
+- You'd like to specify reviwers per author
```
Comment (griffithsbs) [inline README.md:14]: > reviwers
`reviewers`

File: README.md [hunk 2]
```diff
@@ -23,6 +24,7 @@ Code owners own or are responsible for the code in their spaces while reviewers
 This GitHub Action enables you to:
 
 - Auto-assign reviewers based on files changed
+- Auth-assign reviewers based on the author
```
Comment (griffithsbs) [inline README.md:27]: > Auth

Auto

File: README.md [hunk 3]
```diff
@@ -36,6 +38,26 @@ files:
     - js-lovers
 ```
 
+### Auth-assign reviewers based on the author
```
Comment (griffithsbs) [inline README.md:41]: > Auth

Auto

File: src/reviewer.js [hunk 4]
```diff
@@ -32,26 +32,55 @@ function fetch_other_group_members({ author, config }) {
...
 
   Object.entries(config.files).forEach(([ glob_pattern, reviewers ]) => {
     if (changed_files.some((changed_file) => minimatch(changed_file, glob_pattern))) {
-      matching_reviwers.push(...reviewers);
+      matching_reviewers.push(...reviewers);
     }
   });
 
-  const indivisuals = replace_groups_with_individuals({ reviewers: matching_reviwers, config });
+  const indivisuals = replace_groups_with_individuals({ reviewers: matching_reviewers, config });
```
Comment (griffithsbs) [inline src/reviewer.js:49]: > indivisuals

Individuals

**Review by griffithsbs — COMMENTED** [review]
@necojackarc I noticed a few purely cosmetic typos whilst browsing through this!

**Comment by necojackarc** [comment]
@griffithsbs Thanks! I've fixed them in 3487f24ede2df144881f56049643f72a2ce1490b 👍

**Pull request merged**